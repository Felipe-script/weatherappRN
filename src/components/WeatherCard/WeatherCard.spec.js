import * as React from 'react';
import {WeatherCard} from './index';
import renderer from 'react-test-renderer';

const mainMock = {
  temp: 32,
  feels_like: 30,
  temp_min: 31.83,
  temp_max: 31.83,
  pressure: 1011,
  humidity: 25,
  sea_level: 1011,
  grnd_level: 922,
};

const WeatherMock = {
  id: 802,
  main: 'Clouds',
  description: 'nuvens dispersas',
  icon: '03d',
};

const weatherMock = {
  main: {...mainMock},
  weather: {...WeatherMock},
};
console.log('weatherMock', weatherMock);

it('renders correctly', () => {
  const tree = renderer.create(
    <WeatherCard main={weatherMock.main} weather={weatherMock.weather}>
      test
    </WeatherCard>,
  );
  expect(tree).toMatchSnapshot();
});
