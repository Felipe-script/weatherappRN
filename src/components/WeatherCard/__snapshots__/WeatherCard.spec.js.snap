// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#34495e",
        "borderRadius": 10,
        "display": "flex",
        "flexDirection": "column",
        "height": 300,
        "marginBottom": 20,
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": 20,
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
        "width": 300,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 20,
        },
      ]
    }
  >
    nuvens dispersas
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#ecf0f1",
          "fontSize": 16,
        },
      ]
    }
  >
    agora: 32 ℃
  </Text>
  <Image
    source={
      Object {
        "uri": "https://openweathermap.org/img/wn/03d@4x.png",
      }
    }
    style={
      Array [
        Object {
          "alignSelf": "center",
          "height": 100,
          "marginTop": 30,
          "width": 100,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-around",
          "marginTop": 16,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#bdc3e7",
            "fontSize": 14,
          },
        ]
      }
    >
      humidade: 
      25
      %
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#bdc3e7",
            "fontSize": 14,
          },
        ]
      }
    >
      sensação: 
      30
      ℃
    </Text>
  </View>
</View>
`;
